src/package_source/registry.rs

```diff
@@ -109,4 +111,17 @@ impl PackageSource for Registry {
 
         Ok(PackageContents::from_buffer(data))
     }
+
+    fn fallback_sources(&self) -> anyhow::Result<Vec<PackageSourceId>> {
+        let fallback_registries = self.index()?.config()?.fallback_registries;
+        let mut sources = Vec::new();
+
+        if let Some(fallbacks) = fallback_registries {
+            for fallback in fallbacks {
+                sources.push(PackageSourceId::Git(fallback));
+            }
+        }
```

Comment:
Using `map` and `collect` here would be better

Reply:
Good point, I still find using the iter way of doing things a little weird but it's definitely much cleaner here!